src/Course/List.hs

```diff
@@ -165,12 +163,10 @@ filter =
 -- prop> \x -> (x ++ y) ++ z == x ++ (y ++ z)
 --
 -- prop> \x -> x ++ Nil == x
-(++) ::
-  List a
-  -> List a
-  -> List a
-(++) =
-  error "todo: Course.List#(++)"
+(++) :: List a -> List a -> List a
+-- (++) Nil b = b
+-- (++) (h :. t) l = h :. (t ++ l)
+(++) as bs = foldRight (\a acc -> a :. acc) bs as
```

Comment:
As cons `(:.)` has this type `:: a -> List a -> List a`
you can stick it right into the hole:

```suggestion
(++) as bs = foldRight (:.) bs as
```